sphinx/config.py

```diff
@@ -338,6 +355,20 @@ def eval_config_file(filename: str, tags: Optional[Tags]) -> Dict[str, Any]:
     return namespace
 
 
+def read_config_file_yaml(filename: str, tags: Optional[Tags]) -> Dict[str, Any]:
+    """Evaluate a YAML config file."""
+    try:
+        with open(filename, encoding='utf-8') as fp:
+            namespace = yaml.safe_load(fp)
+    except Exception as exc:
+        msg = __("There is a syntax error in your configuration file:\n\n%s")
+        raise ConfigError(msg % traceback.format_exc()) from exc
+    if not isinstance(namespace, dict):
+        raise ConfigError(__("Configuration file is not a dictionary"))
```

Comment:
minor: It would be better to mention the filename `config.yml` in these error messages. It lets users know what file they should fix.

Note: I know this is a trial implementation. So I marked this as minor.

Reply:
done and added tests for parsing failures